subsys/net/lib/nrf_cloud/src/nrf_cloud_fota_common.c

```diff
@@ -258,13 +258,13 @@ int nrf_cloud_fota_fmfu_apply(void)
 	err = fmfu_fdev_load(fmfu_buf, sizeof(fmfu_buf), fmfu_dev.dev, fmfu_dev.offset);
 	if (err != 0) {
 		LOG_ERR("Failed to apply full modem update, error: %d", err);
-		(void)nrf_modem_lib_init();
 		return err;
 	}
 
 	err = nrf_modem_lib_shutdown();
 	if (err != 0) {
-		LOG_WRN("nrf_modem_lib_shutdown() failed: %d\n", err);
+		LOG_WRN("nrf_modem_lib_shutdown() failed: %d", err);
+		return err;
```

Comment:
the lack of an error return here was intentional, since the modem update was applied.  also, that is why it is a LOG_WRN.
perhaps use a unique error code for this case?

Reply:
I see, it just stood out a bit and didn't have any comment explaining why the retval was ignored. Yeah, I can add add that.

Reply:
this is the one that really needs a unique return value, so that callers of this function will know that the FOTA update was successful (for status reporting) but there was some other modem lib error.

Reply:
Ok, any suggestion on errno and docstring?

-ECANCELED: Modem FOTA was successful but modem failed to shutdown afterwards?

Reply:
I removed the commit for now, ill revisit it when we start looking into re-initialization of the modem after FOTA in a separate effort.

In general I think it would be wise to have callbacks _asking_ the application to reinit / shutdown / change mode of the modem. There are usually other parts of the system that uses the modem that goes haywire when it is shutdown in other parts of the stack.

Reply:
@simensrostad we were asked a while back to change from rebooting after modem FOTA to re-initializing the modem, but reached the conclusion that it is rare that modem FOTA is even done, and problematic to ensure all parts of the system that have done runtime configuration of the modem and the offloaded network stack repeat their initialization. As a result we did not make the change.